WeakAurasOptions/AuthorOptions.lua

```diff
@@ -1145,10 +1145,29 @@ local function mergeOptions(childIndex, merged, toMerge)
   end
 end
 
+local function tablesAreEqual(t1, t2)
+	if t1 == t2 then return true end
+	local ty1 = type(t1)
+	local ty2 = type(t2)
+	if ty1 ~= ty2 then return false end
+	if ty1 ~= "table" then return false end
+
+	for k1, v1 in pairs(t1) do
+		local v2 = t2[k1]
+		if v2 == nil or not tablesAreEqual(v1, v2) then return false end
+	end
+ 
+	for k2, v2 in pairs(t2) do
+		local v1 = t1[k2]
+		if v1 == nil or not tablesAreEqual(v1, v2) then return false end
+	end
+	return true
+end
+
 local function allChoicesAreDefault(data)
   for _, option in ipairs(data.authorOptions) do
     if option.key ~= nil and option.default ~= nil and option.default ~= data.config[option.key] then
-      return false
+		  return tablesAreEqual(option.default, data.config[option.key])
```

Comment:
The loop needs to continue if the values are equal. It should only return false if the values are not equal.
Also your function also checks for equality for non-table values, so the original check "option.default ~= data.config[option.key]" should not be necessary.

We might have a table comparing equal function already somewhere.

Reply:
Gotcha, sorry. Latest commit should (finally) be correct.